enable-kdump/ubuntu-enable-kdump.yaml

```diff
@@ -0,0 +1,82 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: enable-kdump
+  labels:
+    app: enable-kdump
+spec:
+  selector:
+    matchLabels:
+      name: enable-kdump
+  template:
+    metadata:
+      labels:
+        name: enable-kdump
+    spec:
+      nodeSelector:
+        "cloud.google.com/gke-os-distribution": "ubuntu"
+        "enable-kdump": "true"
+      hostPID: true
+      containers:
+        - name: enable-kdump
+          image: debian
+          imagePullPolicy: Always
+          securityContext:
+            privileged: true
+          command:
+            - /usr/bin/nsenter
+            - -t 1
+            - -m
+            - -u
+            - -i
+            - -n
+            - -p
+            - --
+            - /bin/bash
+            - -c
+            - |
+
+              set -o errexit
+              set -o pipefail
+              set -o nounset
+
+              function check_kdump() {
+                  local kdump_show
+                  kdump_show=$(kdump-config show)
+                  if echo "${kdump_show}" | grep -q "ready to kdump"; then
+                      echo "ready to kdump!"
+                  else
+                      echo "kdump not setup, isn't ready"
+                  fi
+                  echo "${kdump_show}"
+              }
+
+              function install() {
+                  echo "installing kdump"
+                  apt-get update
+                  DEBIAN_FRONTEND=noninteractive apt-get install -y kdump-tools
```

Comment:
I tried the defaults once mistakenly for kdump-tools and the reboot was borked. I believe the first question answered yes causes an interactive reboot (with username/password). when i answered no to the first question, then it does work. did you test this?

Reply:
stack overflow is particularly unhelpful on this question...

Reply:
[This](https://ubuntu.com/server/docs/kernel-crash-dump) might work (modify files after defaults):

```
# File /etc/default/kexec
# Load a kexec kernel (true/false)
LOAD_KEXEC=false

# File /etc/default/kdump-tools
USE_KDUMP=1
```

Reply:
The docs for setting up kdump for ubuntu are here: https://ubuntu.com/server/docs/kernel-crash-dump

The two prompts it shows are first if you want to hook up kexec-tools for future reboots and if kdump-tools should be enabled by default. Ubuntu docs say yes should be answer to both questions. 

From my testing and stackoverflow searching, the `DEBIAN_FRONTEND=noninteractive` makes it so the prompts aren't shown and the -y should answer yes to all the prompts.

I tested it and on the next reboot, `kdump-config show`, showed "ready to dump", and `/proc/cmdline` had the reserved memory for 

```
$ cat /proc/cmdline
BOOT_IMAGE=/boot/vmlinuz-5.0.0-1032-gke root=PARTUUID=f56597fd-2626-41aa-a805-9572e5aced4a ro scsi_mod.use_blk_mq=Y console=ttyS0 net.ifnames=0 crashkernel=512M-:192M
```

based on that I assumed it's setup correctly :). We can trigger a panic and ensure the dump is generated to double check as a test.

Reply:
From my understanding kdump uses kexec to boot a capture kernel. So by making kexec and kdump both handle reboots, are we causing a potential conflict?

Can you try to take a capture on a GKE Node following your script and see if you can load the vmcore using `crash`?

Reply:
yeah, as I understand from [1] and [2] the way kdump works is to loads the capture kernel ("dump capture kernel"). When a crash happens, as I understand the kdump uses kexec to boot up the capture kernel which captures the dump. As a result, as I understand I think both need to enabled here... let me try full test with the crash tool.   

[1] https://wiki.archlinux.org/index.php/Kdump
[2] https://www.kernel.org/doc/html/latest/admin-guide/kdump/kdump.html

Reply:
Were you able to get it tested?

Reply:
Yup, I tested it by triggering crash with sysrq and used `crash` tool to look at the resulting dump. I added a README file to document the process a bit as well.